src/main/java/bio/terra/cli/serialization/userfacing/UserFacing.java

```diff
@@ -0,0 +1,6 @@
+package bio.terra.cli.serialization.userfacing;
+
+public interface UserFacing {
```

Comment:
Not all UF classes are printable currently, so arguably there should be a narrower interface like `UserFacingPrintable` that only the printing ones use, and no default impl here. I feel like there should be some kind of interface to tie these classes together, for example if you had a list of them and wanted to print them all in sequence.

Reply:
I agree the `UF` classes that implement a `print()` method are similar in that they are user-facing output classes (the classes that don't have `print()` I think are all user-facing input classes). But I'm on the fence about the usefulness of this interface, because I can't think of a situation when we'd want to use a UF class without knowing what type it was (i.e. and have to rely on the interface to know what its behavior is). There are situations when we have a `UFResource` and don't know which sub-class it is, but that already uses inheritance and is solvable by defining an abstract method on the base class. Thoughts?

Reply:
The initial smell was that we had a bunch of classes implementing the same method with the same semantics, which means there's an unwritten abstraction somewhere.

Maybe an interface like `UserFacingOutput` or something would make sense for those that print, and I'd remove the default implementation. You could make a purely decorative interface for the input classes, but with no methods in it it would be kind of odd.